### PR TITLE
Improve link buttons and inline link buttons styling

### PR DESCRIPTION
**Pull request by luixxiul**
This PR:

- Remove the default spacing setting from the inline link button
- Replace `kind="link_inline"` of link buttons not embedded in a sentence with `kind="link"`. If spacing is required from the design viewpoint, the link button should be styled individually.

Inline link buttons should not have padding around them by default, as there are languages which do not have the concept of white space characters in a sentence to begin with, Chinese and Japanese for example. It should be decided by translators whether a inline link button needs spacing or not.

Review per commits is recommended.

Signed-off-by: Suguru Hirahara <luixxiul@users.noreply.github.com>

type: task

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

Preview: https://pr8238--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/toasts/AnalyticsToast.tsx [hunk 1]
```diff
@@ -98,7 +98,7 @@ const getAnonymousDescription = (): ReactNode => {
         },
         {
             "UsageDataLink": (sub) => (
-                <AccessibleButton kind="link" onClick={onUsageDataClicked}>{ sub }</AccessibleButton>
+                <AccessibleButton kind="link_inline" onClick={onUsageDataClicked}>{ sub }</AccessibleButton>
```
Comment (luixxiul) [inline src/toasts/AnalyticsToast.tsx:101]: See above: `"Send <UsageDataLink>anonymous usage data</UsageDataLink> which helps us improve %(brand)s. "`

File: src/toasts/AnalyticsToast.tsx [hunk 2]
```diff
@@ -136,7 +136,7 @@ export const showPseudonymousAnalyticsOptInToast = (analyticsOptIn: boolean): vo
         // The user had no analytics setting previously set, so we just need to prompt to opt-in, rather than
         // explaining any change.
         const learnMoreLink = (sub) => (
-            <AccessibleButton kind="link" onClick={onLearnMoreNoOptIn}>{ sub }</AccessibleButton>
+            <AccessibleButton kind="link_inline" onClick={onLearnMoreNoOptIn}>{ sub }</AccessibleButton>
```
Comment (luixxiul) [inline src/toasts/AnalyticsToast.tsx:139]: See below: `"Share anonymous data to help us identify issues. Nothing personal. No third parties. " +
                "<LearnMoreLink>Learn More</LearnMoreLink>", {}, { "LearnMoreLink": learnMoreLink }),`

File: src/components/views/voip/CallView.tsx [hunk 3]
```diff
@@ -429,7 +429,7 @@ export default class CallView extends React.Component<IProps, IState> {
                         transferee: transfereeName,
                     },
                     {
-                        a: sub => <AccessibleButton kind="link" onClick={this.onTransferClick}>
+                        a: sub => <AccessibleButton kind="link_inline" onClick={this.onTransferClick}>
```
Comment (luixxiul) [inline src/components/views/voip/CallView.tsx:432]: See above: `"Consulting with %(transferTarget)s. <a>Transfer to %(transferee)s</a>",`

File: res/css/views/elements/_AccessibleButton.scss [hunk 4]
```diff
@@ -137,15 +137,11 @@ limitations under the License.
...
         line-height: inherit;
-    }
-
-    &.mx_AccessibleButton_kind_link {
         padding: 0;
     }
 
     &.mx_AccessibleButton_kind_link_inline {
         display: inline;
-        padding: 0 2px;
```
Comment (luixxiul) [inline res/css/views/elements/_AccessibleButton.scss:148]: There is a lot of declarations to cancel the inline 2px padding, and it is not quite sure what this has been introduced for.

File: res/css/views/messages/_ViewSourceEvent.scss [hunk 5]
```diff
@@ -36,7 +36,6 @@ limitations under the License.
         visibility: hidden;
         // override styles from AccessibleButton
         border-radius: 0;
-        padding: 0;
```
Comment (luixxiul) [inline res/css/views/messages/_ViewSourceEvent.scss:39]: `.mx_ViewSourceEvent_toggle` is applied to `AccessibleButton` with `kind='link'` on `ViewSourceEvent.tsx`.

File: res/css/structures/_RoomDirectory.scss [hunk 6]
```diff
@@ -180,21 +180,13 @@ limitations under the License.
 .mx_RoomDirectory > span {
     font-size: $font-15px;
     margin-top: 0;
-
-    .mx_AccessibleButton {
-        padding: 0;
```
Comment (luixxiul) [inline res/css/structures/_RoomDirectory.scss:185]: For `create a new room` link button.

File: res/css/views/dialogs/_SpotlightDialog.scss [hunk 7]
```diff
@@ -468,10 +467,6 @@ limitations under the License.
 
         > span {
             align-self: center;
-
-            .mx_AccessibleButton_kind_link_inline {
-                padding: 0;
-            }
```
Comment (luixxiul) [inline res/css/views/dialogs/_SpotlightDialog.scss:474]: after1

File: res/css/views/dialogs/_InviteDialog.scss [hunk 8]
```diff
@@ -256,10 +256,6 @@ limitations under the License.
 
 .mx_InviteDialog_helpText {
     margin: 0;
-
-    .mx_AccessibleButton_kind_link {
-        padding: 0;
-    }
```
Comment (luixxiul) [inline res/css/views/dialogs/_InviteDialog.scss:262]: `AccessibleButton` with `kind=link` is not used here.

after3

**Review by turt2live — APPROVED** [review]
let's see what happens :D

**Pull request merged**